File: Java/src/session3/td/ConsIter.java
package session3.td;

public class ConsIter implements Mot{
	private char caractere;
	private Mot reste;

	public ConsIter(char i, Mot ens) {
		this.caractere = i;
		this.reste = ens;
	}

	@Override
	public boolean casCons() {
		return true;
	}
	
	@Override
	public int taille() {
		int result = 0;
		Mot tempRest = reste;
		while(!tempRest.casVide()) {
			result++;
			tempRest = tempRest.reste();
		}
		return 1 + result;
		
	}

	@Override
	public char caractere() {
		return this.caractere;
	}

	@Override
	public Mot reste() {
		return this.reste;
	}

	/*@Override
	public char caractereA(int index) {
		if(index<this.taille()) {
			int result = 0;
			char resultChar = ' ';
			Mot tempRest = reste;
			while(!tempRest.casVide()) {
				if(result==index-1) {
					resultChar = tempRest.caractere();
					break;
				}
				result++;
				tempRest = tempRest.reste();
			}
			return resultChar;
		}else {
			throw new UnsupportedOperationException();
		}
		
	}*/

	@Override
	public int derniereOccurence(char ch) {
			int i = 0;
			int occurence = 0;
			Mot tempRest = reste;
			while(!tempRest.casVide()) {
				if(tempRest.caractere()==ch) {
					occurence = i+1;
				}
				i++;
				tempRest = tempRest.reste();
			}
			return occurence;
	}
}

File: TD3/typescript/session3/td/ConsIter.ts
import { Mot } from "./Mot";

export class ConsIter extends Mot{

    public var_lettre:string;
    public var_reste:Mot;

    public constructor(i:string, ens:Mot){
        super();
        this.var_lettre=i;
        this.var_reste=ens;
    }

    casCons():boolean{
        return true;
    }

    taille(): number {
        var result = 0;
		var tempRest:Mot = this.var_reste;
		while(!tempRest.casVide()) {
			result++;
			tempRest = tempRest.reste();
		}
        return 1 + result;
    }

    caractere():string{
        return this.var_lettre;
    }

    reste():Mot{
        return this.var_reste;
    }

}